execution-core/src/lib.rs

```diff
@@ -65,6 +65,15 @@ pub mod plonk {
     };
 }
 
+/// Groth16 circuitry
+#[cfg(feature = "zk")]
+pub mod groth16 {
+    pub use ark_bn254 as bn254;
+    pub use ark_groth16::*;
```

Comment:
I am heavily opinionated against re-exports with an `*`, although I understand that you do it because the `ark_groth16` types are rather extensive.
Do you think we can agree to not re-export groth16 types via execution-core? After all our protocol operates on plonk and afaict groth16 is only needed for the bridge.

Reply:
Our protocol, with the addition of the `verify_groth16` host function, now also includes groth16. "The protocol" includes everything that contracts can do, not only the transactions themselves, so I feel this actually does belong in `execution-core`.

As far as re-exports with `*`... notice I'm being very deliberate here. `*` is used for the `ark-groth16` crate, and all others are included as just a module, stripping their `ark-` prefix. Unless we want to import types using something like `execution_core::groth16::groth16::Groth16`, I suggest keeping it like this. Besides, it's all neatly encapsulated in `execution_core::groth16`.

Reply:
I am not convinced that this justifies a re-export like this and would still prefer to not add groth types to `execution-core`.
If you do choose to add them, then please do so without a wildcard and separate the `zk` feature so that we don't add `groth16` dependencies when we want to use plonk only.

Reply:
I'll add a `plonk` and a `groth16` feature. I'm seeing no reason why `*` imports are not allowed - especially under such a neatly scoped module. Could you justify why? And what is the alternative?